File: mc-plugins/moreutils/src/main/java/org/moreutils/utils/TryTrie.java
package org.moreutils.utils;

/**
 * @Author: Yangzhengqian
 * @Description:
 * @Date:Created time 2019/12/30 14:31
 * @Modified By:
 */
public class TryTrie {
    public static void main(String[] args) {
        Trie trie=new Trie();
        trie.add("apple");
        System.out.println(trie.containsPrefix("app"));
    }
}

File: mcplugins-kt/moreutils/src/main/kotlin/org/moreutils/utils/TryTrie.kt
package org.moreutils.utils

/**
 * @Author: Yangzhengqian
 * @Description:
 * @Date:Created time 2019/12/30 14:31
 * @Modified By:
 */
object TryTrie {
    @JvmStatic
    fun main(args: Array<String>) {
        val trie = Trie()
        trie.add("apple")
        println(trie.containsPrefix("app"))
    }
}